### PR TITLE
[Profile] Check duplicated info in profiles

### PR DESCRIPTION
**Pull request by nxqbao**
### Description

- Check whether each info in the new profile is duplicated with the existing ones, and cannot be reused when changed.

**Note for mainnet migration, the migration steps will be as follows:**
- Create a `initializeVx` method for adding profiles for all current candidates, not allow anyone to add profile until this `initializeVx` is called. The info of the adding profiles are hardcoded.
- Expose a `updatePubkey` method to allow candidate update their profile.

### CODE REVIEW

**Review by minh-bq — APPROVED** [review]

File: contracts/ronin/profile/ProfileStorage.sol [hunk 1]
```diff
@@ -22,14 +24,69 @@ abstract contract ProfileStorage is IProfile, HasContracts {
...
+
+  function _checkDuplicatedGovernor(address governor) internal view {
+    if (_registry[uint256(uint160(governor))]) {
+      revert ErrDuplicatedInfo("governor", uint256(uint160(governor)));
+    }
+  }
+
+  function _checkDuplicatedPubkey(bytes memory pubkey) internal view {
+    if (_registry[_hashPubkey(pubkey)]) {
+      revert ErrDuplicatedInfo("pubkey", _hashPubkey(pubkey));
```
Comment (ducthotran2010) [inline contracts/ronin/profile/ProfileStorage.sol:86]: Not sure this info is helpful to debug, should we emit `pubkey` instead?

File: contracts/ronin/profile/Profile.sol [hunk 2]
```diff
@@ -38,12 +38,30 @@ contract Profile is IProfile, ProfileStorage, Initializable {
...
+   */
+  function changePubkey(address id, bytes memory pubkey) external {
+    CandidateProfile storage _profile = _getId2ProfileHelper(id);
+    if (msg.sender != _profile.admin) revert ErrUnauthorized(msg.sig, RoleAccess.ADMIN);
+    _checkDuplicatedPubkey(pubkey);
+
+    _registry[_hashPubkey(_profile.pubkey)] = false;
+
+    _profile.pubkey = pubkey;
+    _registry[_hashPubkey(_profile.pubkey)] = true;
```
Comment (ducthotran2010) [inline contracts/ronin/profile/Profile.sol]:
```suggestion
    _registry[_hashPubkey(pubkey)] = true;
```

File: contracts/interfaces/IProfile.sol [hunk 3]
```diff
@@ -30,11 +30,20 @@ interface IProfile {
   event ProfileAdded(address indexed id);
   /// @dev Event emitted when a address in a profile is changed.
   event ProfileAddressChanged(address indexed id, RoleAccess indexed addressType);
+  /// @dev Event emitted when the pubkey of the `id` is changed.
+  event PubkeyChanged(address indexed id);
```
Comment (ducthotran2010) [inline contracts/interfaces/IProfile.sol]: Should we emit the new pubkey?

File: contracts/ronin/profile/ProfileStorage.sol [hunk 4]
```diff
@@ -22,14 +24,69 @@ abstract contract ProfileStorage is IProfile, HasContracts {
...
+    }
+  }
+
+  function _checkDuplicatedGovernor(address governor) internal view {
+    if (_registry[uint256(uint160(governor))]) {
+      revert ErrDuplicatedInfo("governor", uint256(uint160(governor)));
+    }
+  }
+
+  function _checkDuplicatedPubkey(bytes memory pubkey) internal view {
```
Comment (ducthotran2010) [inline contracts/ronin/profile/ProfileStorage.sol:84]: Missing comments

**Review by ducthotran2010 — APPROVED** [review]
Only info issues

File: contracts/ronin/profile/ProfileStorage.sol (hunk 4 above)
Comment (nxqbao) [reply contracts/ronin/profile/ProfileStorage.sol:84]: I consider the comments are scary noise comment (See Page 64 of [this book](https://books.google.com.vn/books?redir_esc=y&id=_i6bDeoCQzsC&q=noise+comment#v=onepage&q&f=false)), so I intentionally left it blank.

**Pull request merged**